### ISSUE TITLE
Users encountering an error while attempting to pay for order using UPE method

### ISSUE THREAD

**Issue by timur27** [Automattic/woocommerce-payments#5808]
### Describe the bug
When trying to Pay for order with any of the available UPE methods, user sees an error _**Invalid payment method. Please input a new card number.**_

### To Reproduce

Enable split UPE and:
1. Buy a product using the shortcode checkout and UPE method (e.g. Sofort) 
2. On the third-party test payment page, choose _**Fail test payment**_
3. After redirected back to the checkout page, go to _**My account -> Orders**_, find the failed order and click _**Pay**_
4. On the _**Pay for order**_ page, choose any of the available UPE methods and click _**Pay for order**_

### Actual behavior
Users are seeing _**Invalid payment method. Please input a new card number.**_ error notice. 

### Screenshots

![image.png](https://images.zenhubusercontent.com/59de7a6cb0222d5de47b4727/62610a52-2727-4392-a270-c89f379009c7)

### Expected behavior

Users shouldn be able to pay for order successfully, using any of the available payment method on the list.

### Desktop (please complete the following information):

* OS: MacOS
* Browser: Chrome

**Comment by timur27** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Heisenberg-63586f2ec60359001310fa6a/issues/Automattic/woocommerce-payments/5808?planning-poker) with Zenhub @FangedParakeet @mdmoore

**Comment by FangedParakeet** [comment]
I now have a PR ready that will (eventually) resolve this issue, but I am going to mark this issue temporarily blocked for now, since I have also learned that this functionality is also blocked by #5854 and we will not be able to return this functionality, until a hero appears to save us from despair by resolving that particular issue.

> I have identified three issues that are causing these frustrating failures.
>
>1. If order currency and customer selected currency do not match, if the customer selected currency is not supported by available payment method, payment intent will produce an error and payment elements will fail to mount.
>2. Pay for order form incorrectly detects when WCPay payment method is selected at checkout.
>3. Payment intent confirms successfully, but process_redirect_payment function exits early, order is not updated, and instead marked as a failure.

Please see #6177 for more specific details on these maladies and their various cures.
